Improve logging in the TSConverter

jiashenC opened the pull request:
Summary: Currently, running explain with TORCH_LOGS enabled will cause duplicate loggings because explain uses the exact same code path for covnersion. This PR just disables logging when it is running explain. And move all logging to convert() to prevent from logging from __init__ when we are just using explain.

Test Plan: Manual testing with attached outputs.

Reviewed By: SherlockNoMad, angelayi

Differential Revision: D60199007

Skylion007 commented on torch/_export/converter.py:
```diff
@@ -1015,6 +1013,17 @@ def __init__(
+DEBUG: (TORCH_LOGS="+export" <cmd>), additionally
+    * Log conversion IR by IR in a format of [<conversion handler name>] converts [<IR>].
+        """)
+        log.info(f"TorchScript graph\n\n{self.ts_graph}\n")  # noqa: G004
```
This seems expensive to log:
```suggestion
        log.info(f"TorchScript graph\n\n%s\n", {self.ts_graph})
```

Commit:
Improve logging in the TSConverter (#132082)

Summary:
Currently, running explain with TORCH_LOGS enabled will cause duplicate loggings because explain uses the exact same code path for covnersion. This PR just disables logging when it is running explain. And move all logging to convert() to prevent from logging from __init__ when we are just using explain.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/132082

Test Plan: Manual testing with attached outputs.

Reviewed By: ydwu4, angelayi

Differential Revision: D60199007

Pulled By: jiashenC
